### PR TITLE
doc: Add references to the terraform modules in the resources documentations

### PR DESCRIPTION
**Pull request by rubenVB01**
## Description

Added references to the push_based_log_export, encryption_at_rest, privatelink_endpoint and privatelink_endpoint_service resources to their respective modules.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: docs/resources/encryption_at_rest.md [hunk 1]
```diff
@@ -6,6 +6,8 @@
 [Azure Key Vault](https://docs.atlas.mongodb.com/security-azure-kms/#security-azure-kms)
 [Google Cloud KMS](https://docs.atlas.mongodb.com/security-gcp-kms/#security-gcp-kms)
 
+An [encryption at rest terraform module](https://registry.terraform.io/modules/terraform-mongodbatlas-modules/encryption-at-rest/mongodbatlas/latest) makes use of this resource and simplifies its usages.
```
Comment (EspenAlbert) [inline docs/resources/encryption_at_rest.md]:
```suggestion
The [encryption at rest Terraform module](https://registry.terraform.io/modules/terraform-mongodbatlas-modules/encryption-at-rest/mongodbatlas/latest) makes use of this resource and simplifies its usages.
```

File: docs/resources/privatelink_endpoint.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 `mongodbatlas_privatelink_endpoint` provides a Private Endpoint resource. This represents a [Private Endpoint Service](https://www.mongodb.com/docs/atlas/security-private-endpoint/#private-endpoint-concepts) that can be created in an Atlas project.
 
+A [private link terraform module](https://registry.terraform.io/modules/terraform-mongodbatlas-modules/private-endpoint/mongodbatlas/latest) makes use of this resource and simplifies its usages.
```
Comment (EspenAlbert) [inline docs/resources/privatelink_endpoint.md]:
```suggestion
The [private link Terraform module](https://registry.terraform.io/modules/terraform-mongodbatlas-modules/private-endpoint/mongodbatlas/latest) makes use of this resource and simplifies its usages.
```

File: docs/resources/push_based_log_export.md [hunk 3]
```diff
@@ -3,6 +3,8 @@
 `mongodbatlas_push_based_log_export` provides a resource for push-based log export feature. The resource lets you configure, enable & disable the project level settings for the push-based log export feature. Using this resource you 
 can continually push logs from mongod, mongos, and audit logs to an Amazon S3 bucket. Atlas exports logs every 5 minutes.
 
+A [push based log export terraform module](https://registry.terraform.io/modules/terraform-mongodbatlas-modules/push-based-log-export/mongodbatlas/latest) makes use of this resource and simplifies its usages.
```
Comment (EspenAlbert) [inline docs/resources/push_based_log_export.md]:
```suggestion
The [push based log export Terraform module](https://registry.terraform.io/modules/terraform-mongodbatlas-modules/push-based-log-export/mongodbatlas/latest) makes use of this resource and simplifies its usages.
```

File: docs/resources/privatelink_endpoint_service.md [hunk 4]
```diff
@@ -2,6 +2,8 @@
 
 `mongodbatlas_privatelink_endpoint_service` provides a Private Endpoint Interface Link resource. This represents a Private Endpoint Interface Link, which adds one [Interface Endpoint](https://www.mongodb.com/docs/atlas/security-private-endpoint/#private-endpoint-concepts) to a private endpoint connection in an Atlas project.
 
+A [private link terraform module](https://registry.terraform.io/modules/terraform-mongodbatlas-modules/private-endpoint/mongodbatlas/latest) makes use of this resource and simplifies its usages.
```
Comment (EspenAlbert) [inline docs/resources/privatelink_endpoint_service.md]:
```suggestion
The [private link Terraform module](https://registry.terraform.io/modules/terraform-mongodbatlas-modules/private-endpoint/mongodbatlas/latest) makes use of this resource and simplifies its usages.
```

**Review by EspenAlbert — APPROVED** [review]
Some nit, but maybe docs team have better suggestions

File: templates/resources/push_based_log_export.md.tmpl [hunk 5]
```diff
@@ -3,6 +3,8 @@
 `{{.Name}}` provides a resource for push-based log export feature. The resource lets you configure, enable & disable the project level settings for the push-based log export feature. Using this resource you 
 can continually push logs from mongod, mongos, and audit logs to an Amazon S3 bucket. Atlas exports logs every 5 minutes.
 
+A [push based log export terraform module](https://registry.terraform.io/modules/terraform-mongodbatlas-modules/push-based-log-export/mongodbatlas/latest) makes use of this resource and simplifies its usages.
```
Comment (jwilliams-mongo) [inline templates/resources/push_based_log_export.md.tmpl]:
```suggestion
The [push based log export Terraform module](https://registry.terraform.io/modules/terraform-mongodbatlas-modules/push-based-log-export/mongodbatlas/latest) makes use of this resource and simplifies its use.
```

File: docs/resources/push_based_log_export.md (hunk 3 above)
Comment (jwilliams-mongo) [inline docs/resources/push_based_log_export.md]:
```suggestion
A [push based log export terraform module](https://registry.terraform.io/modules/terraform-mongodbatlas-modules/push-based-log-export/mongodbatlas/latest) makes use of this resource and simplifies its use.
```

File: docs/resources/privatelink_endpoint_service.md (hunk 4 above)
Comment (jwilliams-mongo) [inline docs/resources/privatelink_endpoint_service.md]:
```suggestion
A [private link terraform module](https://registry.terraform.io/modules/terraform-mongodbatlas-modules/private-endpoint/mongodbatlas/latest) makes use of this resource and simplifies its use.
```

File: docs/resources/privatelink_endpoint.md [hunk 6]
```diff
@@ -79,4 +81,4 @@ Private Endpoint Service can be imported using project ID, private link ID, prov
 $ terraform import mongodbatlas_privatelink_endpoint.test 1112222b3bf99403840e8934-3242342343112-AWS-us-east-1
 ```
 
-See detailed information for arguments and attributes: [MongoDB API Private Endpoint Service](https://docs.atlas.mongodb.com/reference/api/private-endpoints-service-create-one//)
\ No newline at end of file
+See detailed information for arguments and attributes: [MongoDB API Private Endpoint Service](https://docs.atlas.mongodb.com/reference/api/private-endpoints-service-create-one//)
```
Comment (jwilliams-mongo) [inline docs/resources/privatelink_endpoint.md]:
```suggestion
See detailed information for arguments and attributes: [MongoDB API Private Endpoint Service](https://docs.atlas.mongodb.com/reference/api/private-endpoints-service-create-one/)
```

File: docs/resources/privatelink_endpoint.md (hunk 2 above)
Comment (jwilliams-mongo) [inline docs/resources/privatelink_endpoint.md]:
```suggestion
A [private link terraform module](https://registry.terraform.io/modules/terraform-mongodbatlas-modules/private-endpoint/mongodbatlas/latest) makes use of this resource and simplifies its use.
```

File: docs/resources/encryption_at_rest.md (hunk 1 above)
Comment (jwilliams-mongo) [inline docs/resources/encryption_at_rest.md]: 'use' is simpler.
```suggestion
An [encryption at rest terraform module](https://registry.terraform.io/modules/terraform-mongodbatlas-modules/encryption-at-rest/mongodbatlas/latest) makes use of this resource and simplifies its use.
```

**Review by AgustinBettati — APPROVED** [review]
LGTM

**Review by oarbusi — APPROVED** [review]

**Review by jwilliams-mongo — APPROVED** [review]

**Pull request merged**